packages/core/test/financial-templates/common/financial-product-libraries/expiring-multiparty-libraries/PreExpirationIdentifierTransformationFinancialProductLibrary.js

```diff
@@ -1,98 +1,110 @@
+    identifierTransformationFPL = await PreExpirationIdentifierTransformationFinancialProductLibrary.new().send({
+      from: accounts[0],
+    });
     expiringMultiParty = await ExpiringMultiPartyMock.new(
-      identifierTransformationFPL.address,
+      identifierTransformationFPL.options.address,
       expirationTime,
       { rawValue: collateralizationRatio.toString() },
       priceFeedIdentifier,
-      timer.address
-    );
+      timer.options.address
+    ).send({ from: accounts[0] });
 
-    await identifierTransformationFPL.setFinancialProductTransformedIdentifier(
-      expiringMultiParty.address,
-      transformedPriceFeedIdentifier
-    );
+    await identifierTransformationFPL.methods
+      .setFinancialProductTransformedIdentifier(expiringMultiParty.options.address, transformedPriceFeedIdentifier)
+      .send({ from: accounts[0] });
   });
   it("Transformation correctly set", async () => {
     assert.equal(
       hexToUtf8(
-        await identifierTransformationFPL.getTransformedIdentifierForFinancialProduct(expiringMultiParty.address)
+        await identifierTransformationFPL.methods
+          .getTransformedIdentifierForFinancialProduct(expiringMultiParty.options.address)
+          .call()
       ),
       hexToUtf8(transformedPriceFeedIdentifier)
     );
   });
   it("Can not re-set the transformation for a given financial product", async () => {
     assert(
       await didContractThrow(
-        identifierTransformationFPL.setFinancialProductTransformedIdentifier(
-          expiringMultiParty.address,
-          transformedPriceFeedIdentifier
-        )
+        identifierTransformationFPL.methods
+          .setFinancialProductTransformedIdentifier(expiringMultiParty.options.address, transformedPriceFeedIdentifier)
+          .send({ from: accounts[0] })
       )
     );
   });
   it("Correctly transforms the identifier pre-expiration", async () => {
     // Calling the transformation function through the emp mock.
     assert.equal(
       hexToUtf8(
-        await expiringMultiParty.transformPriceIdentifier((await expiringMultiParty.getCurrentTime()).toString())
+        await expiringMultiParty.methods
+          .transformPriceIdentifier((await expiringMultiParty.methods.getCurrentTime().call()).toString())
+          .call()
       ),
       hexToUtf8(transformedPriceFeedIdentifier)
     );
 
     // Calling the transformation function as a mocked emp caller should also work.
     assert.equal(
       hexToUtf8(
-        await expiringMultiParty.transformPriceIdentifier.call((await expiringMultiParty.getCurrentTime()).toString(), {
-          from: expiringMultiParty.address,
-        })
+        await expiringMultiParty.methods
+          .transformPriceIdentifier((await expiringMultiParty.methods.getCurrentTime().call()).toString())
+          .call({ from: expiringMultiParty.options.address })
```

Comment:
it's a bit painful that the `address` prop is now within the contract instances `options` section

Reply:
Yeah, not sure what we can do about that. I think there's a `._address` private field we could use, but I'm not sure that's recommended.